### PR TITLE
Add important elements of target architecture

### PR DESCRIPTION
**Pull request by stephenkaye-ho**
Link to reference architecture doesn't appear to be followed so we've extracted highlights of the reference architecture into this page for the convenience of the reader

### CODE REVIEW

File: documentation/architecture/index.md [hunk 1]
```diff
@@ -9,30 +9,103 @@ tags:
...
+
+Security groups are recommended as the primary data-plane isolation mechanism between applications that may coexist in the same VPC. It is anticipated that unrelated applications would coexist in their respective tiers without ever permitting east-west traffic flows.
+
+The following subnets will be defined:
+
+* **TGW subnet:** This subnet hosts the elastic-network interfaces for the TGW attachment. A /27 subnet is sufficient.
+* **Web subnet:** This subnet hosts front-end or otherwise 'client' facing infrastructure. A /20 or larger subnet is recommended to facilitate auto-scaling.
+* **App subnet:** This subnet hosts app-tier code (EC2, containers, etc). A /19 or larger subnet is recommended to facilitate auto-scaling.
+* **Data subnet:** This subnet hosts data-tier code (RDS instances, ElastiCache instances). A /21 or larger subnet is recommended.
+* **Mgmt subnet:** This subnet hosts bastion or other management instances. A /21 or larger subnet is recommended.
```
Comment (LiamMacP) [inline documentation/architecture/index.md:98]:
```suggestion
* **Management subnet:** This subnet hosts bastion or other management instances. A /21 or larger subnet is recommended.
```

While accepted as a shorthand, I think for this kind of documentation it reads better as longhand?

Comment (danielpalmeribm) [reply documentation/architecture/index.md:98]: > While accepted as a shorthand, I think for this kind of documentation it reads better as longhand?

Agree, although if this is just to make the AWS documentation available on our pages, AWS refer to it as a Mgmt subnet in their architecture docs.

Comment (stephenkaye-ho) [reply documentation/architecture/index.md:98]: I think it's also because that's the actual name of the subnet, I'll double check.

Comment (stephenkaye-ho) [reply documentation/architecture/index.md:98]: Checked this and it creates a subnet per VPC but the name is Mgmt e.g. Dev-Mgmt-A, Central-Mgmt-A. @LiamMacP are you happy to resolve the conversation?

Comment (LiamMacP) [reply documentation/architecture/index.md:98]: Yep resolved 👍🏽

**Pull request merged**